js/src/Lines.ts

```diff
@@ -163,173 +165,171 @@ export class Lines extends Mark {
     // Updating the style of the curve, stroke, colors, dashed etc...
     // Could be fused in a single function for increased readability
     // and to avoid code repetition
-    update_line_style() {
+    update_line_style(): void {
         this.d3el.selectAll(".curve").select(".line")
           .style("stroke-dasharray", _.bind(this.get_line_style, this));
-        if (this.legend_el) {
-            this.legend_el.select("path")
+        if (this.legendEl) {
+            this.legendEl.select("path")
               .style("stroke-dasharray", _.bind(this.get_line_style, this));
         }
     }
 
-    update_stroke_width(model, stroke_width) {
+    update_stroke_width(model, strokeWidth): void {
         this.compute_view_padding();
         this.d3el.selectAll(".curve").select(".line")
-          .style("stroke-width", stroke_width);
-        if (this.legend_el) {
-            this.legend_el.select("path")
-              .style("stroke-width", stroke_width);
+          .style("stroke-width", strokeWidth);
+        if (this.legendEl) {
+            this.legendEl.select("path")
+              .style("stroke-width", strokeWidth);
         }
     }
 
-    update_style() {
-        const that = this,
-            fill = this.model.get("fill"),
-            fill_color = this.model.get("fill_colors"),
-            fill_opacities = this.model.get("fill_opacities");
+    update_style(): void {
+        const fill = this.model.get("fill"),
+            fillColor = this.model.get("fill_colors"),
+            fillOpacities = this.model.get("fill_opacities");
         // update curve colors
         const curves = this.d3el.selectAll(".curve")
         curves.select(".line")
           .style("opacity", this.get_mark_opacity.bind(this))
-          .style("stroke", function(d, i) {
-              return that.get_mark_color(d, i) || fill_color[i];
+          .style("stroke", (d, i) => {
+              return this.get_mark_color(d, i) || fillColor[i];
           })
-          .style("fill", function(d, i) {
-              return fill === "inside" ? that.get_fill_color(d, i) : "";
+          .style("fill", (d, i) => {
+              return fill === "inside" ? this.get_fill_color(d, i) : "";
           })
           .style("fill-opacity", function(d, i) {
-              return fill === "inside" ? fill_opacities[i] : "";
+              return fill === "inside" ? fillOpacities[i] : "";
           });
         curves.select(".area")
-          .style("fill", function(d, i) { return that.get_fill_color(d, i); })
-          .style("opacity", function(d, i) { return fill_opacities[i]; });
+          .style("fill", (d, i) => { return this.get_fill_color(d, i); })
+          .style("opacity", function(d, i) { return fillOpacities[i]; });
```

Comment:
There are still some `function`s around

Reply:
Yes... But I was obliged to use `functions` in some instances. I changed the cases where we were using `that` to `this`.
We sometimes might want to use `function`, right? I don't think it qualifies as bad code per si, it's just related to what you want to do with the scope of your function. So... I'd say it's okay to have them?

Reply:
The thing is that if we need to change that code and call an instance method in there with `this`, we won't use the right context. Just for this and for the sake of homogeneity I would change them all. Let's get rid of all the old JavaScript :P

Reply:
Hum, I see. Okay then.

Reply:
I didn't know it was old JS, thought it was used side by side with arrow functions.